### ISSUE TITLE
Slow image pulling when running AzureStaticWebApp@0 task

### ISSUE THREAD

**Issue by rmaziarka** [Azure/static-web-apps#1173]
**Describe the bug**

When you run AzureStaticWebApp@0 task, it takes 80% of the time to pull the image and 20% to actual push the files to the cloud. In my case:
- Total publish: 1 minute 5 seconds
- Pulling the image: 55 seconds
- Uploading data: 10 seconds

**Expected behavior**
Either:
- Lower the time to publish by avoiding pulling the image?
- Create a different way to handle the upload, maybe by native API?

**Comment by thomasgauvin** [comment]
Hi @rmaziarka, this is something we are investigating. The default AzureStaticWebApp@0 image is intended to be the simplest set up for folks who may not have familiarity with building & configuring GitHub pipelines. 

For you use case, your time breakdown seems to indicate that you do not need to build your project and only need to upload the static assets, correct?

**Comment by rmaziarka** [comment]
> For you use case, your time breakdown seems to indicate that you do not need to build your project and only need to upload the static assets, correct?

Yes, only uploading is currently carried out. I don't use building the project in this stage of pipeline.

**Comment by rmaziarka** [comment]
@thomasgauvin is there an API similar to below, but with sending directly ZIP files to the static site?
https://learn.microsoft.com/en-us/rest/api/appservice/static-sites/create-zip-deployment-for-static-site?tabs=HTTP

**Comment by rmaziarka** [comment]
I see that there is SWA CLI where you can deploy your app from the console:
https://azure.github.io/static-web-apps-cli/docs/cli/swa-deploy/
`swa deploy`

Could you let me know which API is used to do such a deployment? It seems that there is an option to send a ZIP file directly to the SWA.
